From 8904f5452e1776d110b31bdf52b3123205e10fbd Mon Sep 17 00:00:00 2001
From: Derek Foreman <derek.foreman@collabora.com>
Date: Wed, 4 Sep 2024 06:39:31 -0500
Subject: [PATCH 60/94] [PATCH 02/18] dw-hdmi-qp: Add missing capabilities

The hardware can do 10-bit and YUV, so add it here.

Signed-off-by: Derek Foreman <derek.foreman@collabora.com>

Gbp-Pq: Topic hdmi
Gbp-Pq: Name 0002-dw-hdmi-qp-Add-missing-capabilities.patch
---
 drivers/gpu/drm/bridge/synopsys/dw-hdmi-qp.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/bridge/synopsys/dw-hdmi-qp.c b/drivers/gpu/drm/bridge/synopsys/dw-hdmi-qp.c
index b878c4e23..ca292cba4 100644
--- a/drivers/gpu/drm/bridge/synopsys/dw-hdmi-qp.c
+++ b/drivers/gpu/drm/bridge/synopsys/dw-hdmi-qp.c
@@ -680,12 +680,16 @@ struct dw_hdmi_qp *dw_hdmi_qp_bind(struct platform_device *pdev,
 	if (ret)
 		return ERR_PTR(ret);
 
+	hdmi->bridge.max_bpc = 10;
 	hdmi->bridge.driver_private = hdmi;
 	hdmi->bridge.funcs = &dw_hdmi_qp_bridge_funcs;
 	hdmi->bridge.ops = DRM_BRIDGE_OP_DETECT |
 			   DRM_BRIDGE_OP_EDID |
 			   DRM_BRIDGE_OP_HDMI |
 			   DRM_BRIDGE_OP_HPD;
+	hdmi->bridge.supported_formats = BIT(HDMI_COLORSPACE_RGB) |
+					 BIT(HDMI_COLORSPACE_YUV444) |
+					 BIT(HDMI_COLORSPACE_YUV422);
 	hdmi->bridge.of_node = pdev->dev.of_node;
 	hdmi->bridge.type = DRM_MODE_CONNECTOR_HDMIA;
 	hdmi->bridge.vendor = "Synopsys";
-- 
2.46.0

